databases/calendar/calendar.rb

```diff
@@ -12,44 +12,230 @@
+	if days.include?(day)
+		index = days.index(day) + 1
+		plan = db.execute("SELECT events.time, events.event FROM events WHERE events.week_id = ?", [index])
+		plan.each do |event|
+			puts event.join('-')
+		end
+	else
+		puts "I do not recognise that day"
+	end
+end
+
+# allow the user to delete events
+def delete_event(db, days)
+	puts "Would you like to delete an event? (y/n)"
+	if gets.chomp == 'y'
+		puts "What day is the event on?"
+		day = gets.chomp
+		# check for invalid input
+		if days.include?(day)
+			index = days.index(day) + 1
+			puts "Which event would you like to delete from #{day}?"
+			deleted = gets.chomp
+			db.execute("DELETE FROM events WHERE events.event = ? AND events.week_id = ?", [deleted, index])
+		else
+			puts "I do not recognise that day"
+		end
+	end
+end
+
+# remind the user of the important events
+def reminder(db)
+	puts '----------------------'
+	puts "Dont forget to attend:"
+	important = db.execute("SELECT week.day, events.time, events.event FROM week, events WHERE events.week_id = week.id AND events.urgent = 'true'")
+	important.each do |event|
+		puts '----------------------'
+		puts event.join('-')
+	end
+	puts '----------------------'
+end
+
+# make a method to update the time of an event
+def change_time(db, days)
+	puts "Would you like to change the time of an event? (y/n)"
+	if gets.chomp == 'y'
+		puts "What day is the event on?"
+		day = gets.chomp
+		# check for invalid input
+		if days.include?(day)
+			index = days.index(day) + 1
+			puts "What event would you like to change?"
+			event = gets.chomp
+			puts "When will it happen?"
+			time = gets.chomp
+			db.execute("UPDATE events SET time = ? WHERE event = ? AND week_id = ?", [time, event, index])
+		else
+			puts "I do not recognise that day"
+		end
+	end
 end
 
-p db.execute("SELECT events from week WHERE day='Monday'")
+# make something to clear the week and day
+def clear_calendar(days, db, add_events, add_week)
+	puts "Would you like to clear the calendar? (y/n)"
+	if gets.chomp == 'y'
+		puts "Are you sure you would like to clear the calendar? (y/n)"
+		if gets.chomp == 'y'
+			db.execute("DROP TABLE week")
+			db.execute("DROP TABLE events")
+			generate_tables(days, db, add_events, add_week)
+		end
+	end
+end
+
+# driver code
+describe_commands = {
+	'list'=>'access a list of upcoming events',
+	'day'=>'access all the events on a given day',
+	'add'=>'add a new event to your calendar',
+	'remove'=>'remove an item from the calendar',
+	'change'=>'change the time of an event',
+	'clear'=>'remove all items from the calendar'
+}
+
+# tell the user what they can do
+puts "Welcome to your calendar."
+puts "use one of the following commands or type 'done' when finished:"
+describe_commands.each_key do |command|
+	puts "#{command}: #{describe_commands[command]}"
+end
 
+# make the tables (if they're not already there)
+generate_tables(days, db, add_events, add_week)
 
+# let the user select an action
+action = ''
+until action == 'done' do
```

Comment:
This is actually great!  It represents a controller that you will be encountering in web app design.  This is good intuition and I have no complaints!

You could use case if you would like which in the instance runs smoother as it only evaluates the call instead of checking each condition.